nautobot/docs/plugins/region-and-site-data-migration-guide.md

```diff
@@ -0,0 +1,285 @@
+# Region and Site Related Data Model Migration Guide
+
+`Region` and `Site` models are collapsed into `Location` in Nautobot V2.0. As a result, all the `Region` and `Site` related data models are being migrated to use `Location` as well. Below is a comprehensive guide for Nautobot App developers to migrate their `Region` and `Site` related data models to `Location`.
+
+We will be using `ExampleModel` as a relatively simple and hands-on example throughout this guide to better your understanding of the migration process.
+
+## Add Location Fields to Site and Region Related Data Models
+
+`ExampleModel` right now has a `site` ForeignKey field but it does not have a `location` ForeignKey field. We need to add one before we run any other migrations in this guide.
+
+!!! note
+    You can skip this step your data models already have both `site`/`region` fields and a `location` field.
+
+```python
+# models.py
+
+class ExampleModel(OrganizationalModel):
+    site = models.ForeignKey(to="dcim.Site", on_delete=models.PROTECT, blank=True, null=True)
+    name = models.CharField(max_length=20, help_text="The name of this Example.")
+    number = models.IntegerField(default=100, help_text="The number of this Example.")
+    csv_headers = ["name", "number"]
+
+    class Meta:
+        ordering = ["name"]
+...
+```
+
+**DO NOT** delete the `site` ForeignKey field yet.
+
+```python
+# models.py
+
+class ExampleModel(OrganizationalModel):
+    site = models.ForeignKey(to="dcim.Site", on_delete=models.PROTECT, blank=True, null=True)
+    location = models.ForeignKey(to="dcim.Location", on_delete=models.PROTECT, blank=True, null=True)
+    name = models.CharField(max_length=20, help_text="The name of this Example.")
+    number = models.IntegerField(default=100, help_text="The number of this Example.")
+    csv_headers = ["name", "number"]
+
+    class Meta:
+        ordering = ["name"]
+...
+```
+
+Make the migration file by running:
+
+```shell
+nautobot-server makemigrations [app_name] -n [migration_name]
+nautobot-server makemigrations example_plugin -n add_location_field_to_example_model
+```
+
+Apply the migration file by running:
+
+```shell
+nautobot-server migrate [app_name]
+nautobot-server migrate example_plugin
+```
+
+## Create an Empty Migration File to Manually Migrate the Data
```

Comment:
I'm concerned that "manually migrate" will scare people off here. :-) The migration will still be automated, we'll just need to write some code that isn't auto-generated. Perhaps:

```suggestion
## Create an Empty Migration File and Write the Data Migration
```